### PR TITLE
Feature/rbf

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Part 1, the legacy way.

depends on https://github.com/trezor/hd-wallet/pull/51
depends on https://github.com/trezor/connect/commits/feat/rbf

### CODE REVIEW

File: packages/suite/src/actions/suite/trezorConnectActions.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { Dispatch, GetState } from '@suite-types';
 import { isDesktop, isWeb } from '@suite-utils/env';
 import { toTorUrl } from '@suite-utils/tor';
 
-const CONNECT_URL = 'https://connect.trezor.io/8/';
+const CONNECT_URL = 'https://connect.corp.sldev.cz/feat/rbf/';
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/trezorConnectActions.ts]: to be removed after connect release

File: patches/trezor-connect+8.1.18-extended.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
+diff --git a/node_modules/trezor-connect/lib/typescript/account.d.ts b/node_modules/trezor-connect/lib/typescript/account.d.ts
```
Comment (szymonlesisz) [inline patches/trezor-connect+8.1.18-extended.patch]: file will be removed after connect release

File: packages/suite/src/utils/wallet/transactionUtils.ts [hunk 3]
```diff
@@ -256,6 +256,68 @@ export const isTxUnknown = (transaction: WalletAccountTransaction) => {
     );
 };
 
+const getRbfParams = (tx: AccountTransaction, account: Account) => {
+    if (account.networkType !== 'bitcoin') return;
+    // if (tx.type === 'recv' || !tx.rbf || !tx.details || !isPending(tx)) return; // ignore non rbf and mined transactions
+    if (tx.type === 'recv' || !tx.rbf || !tx.details) return; // TEMP: show mined transactions (easier to debug)
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/transactionUtils.ts]: to be removed, just for testing purpose

**Review by slowbackspace — APPROVED** [review]
Couldn't find anything to nitpick on UI part 😞 Good job!

File: packages/suite/src/utils/wallet/transactionUtils.ts [hunk 4]
```diff
@@ -256,6 +256,68 @@ export const isTxUnknown = (transaction: WalletAccountTransaction) => {
...
+                    txid: input.txid!,
+                    vout: input.vout!,
+                    address: addr!.address,
+                    path: addr!.path,
+                    blockHeight: 0,
+                    confirmations: 0,
+                },
+            ];
+        })
+        .filter(input => input !== null);
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/transactionUtils.ts]: not needed filter

File: packages/suite/src/hooks/wallet/useRbfForm.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    // react-hook-form
+    const useFormMethods = useForm<FormState>({ mode: 'onChange', shouldUnregister: false });
+    const { reset, register, setValue, getValues, errors } = useFormMethods;
+
+    // react-hook-form auto register custom form fields (without HTMLElement)
+    useEffect(() => {
+        register({ name: 'outputs', type: 'custom' });
+        register({ name: 'selectedFee', type: 'custom' });
+        register({ name: 'options', type: 'custom' });
```
Comment (szymonlesisz) [inline packages/suite/src/hooks/wallet/useRbfForm.ts:50]: outputs[0].amount

**Comment by tsusanka** [comment]
@szymonlesisz any reason to have this as a Draft?

**Comment by slowbackspace** [comment]
@tsusanka I assume because it really shouldn't get merged in current state as it's waiting for new connect release and removing some code put there only for easier testing. I "approved" only the UI part so it doesn't have to wait on me later.

File: packages/suite/src/utils/wallet/transactionUtils.ts (hunk 4 above)
Comment (szymonlesisz) [reply packages/suite/src/utils/wallet/transactionUtils.ts]: addresses 53c5d9c

**Review by keraf — APPROVED** [review]

File: packages/suite/src/hooks/wallet/useRbfForm.ts (hunk 5 above)
Comment (szymonlesisz) [reply packages/suite/src/hooks/wallet/useRbfForm.ts:50]: addressed 0207bdb and e83dc61

**Review by keraf — APPROVED** [review]

**Review by mroz22 — APPROVED** [review]

**Pull request merged**